### PR TITLE
MATH-1572: Simplify conditional expressions

### PR DESCRIPTION
**Pull request by arturobernalg**
* Simplify bitwise
* Simplify boolean expression
* Simplify conditional expression
* Redundant 'if' statement

### CODE REVIEW

File: commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/fitting/SimpleCurveFitter.java [hunk 1]
```diff
@@ -187,10 +187,7 @@ public int compare(WeightedObservedPoint p1,
                         return comp;
                     }
                     comp = Double.compare(p1.getWeight(), p2.getWeight());
```
Comment (aherbert) [inline commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/fitting/SimpleCurveFitter.java:189]: No need to store the result in `comp`. Just return the result direct.

**Review by aherbert — CHANGES_REQUESTED** [review]
All changes look good.

File: commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/fitting/SimpleCurveFitter.java (hunk 1 above)
Comment (arturobernalg) [reply commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/fitting/SimpleCurveFitter.java]: yes, inline variable.
TY

**Comment by aherbert** [comment]
This is all fine. Can you update the commit message to prefix with MATH-1572.

**Comment by aherbert** [comment]
Closed by commit 1c8d8da63a993c8db554f6afa26c5ea9e8c61b3e

**Pull request closed without merging**